include/inexor/vulkan-renderer/gpu_memory_buffer.hpp

```diff
@@ -1,29 +1,82 @@
 ﻿#pragma once
 
-#include <vma/vma_usage.h>
+#include "vma/vk_mem_alloc.h"
+
+#include <spdlog/spdlog.h>
+
+#include <vulkan/vulkan.h>
 
-#include <cstddef>
 #include <string>
 
 namespace inexor::vulkan_renderer {
 
-/// @brief An abstraction class for the creation of staging buffers.
-/// @note We can't add a std::mutex in here because that would force Buffer to be uncopyable!
-struct Buffer {
-    // TODO: Inherit to make access unified in syntax!
-    // TODO: Map/Unmap memory.
-
+class GPUMemoryBuffer {
+protected:
     std::string name;
 
+    VkDevice device;
+    VmaAllocator vma_allocator;
     VkBuffer buffer = VK_NULL_HANDLE;
-
     VmaAllocation allocation = VK_NULL_HANDLE;
-
     VmaAllocationInfo allocation_info = {};
-
     VkBufferCreateInfo create_info = {};
```

Comment:
The create infos (currently) don't need to be stored for later use.

Reply:
We need this because `StagingBuffer` require this:

```
vertex_buffer_copy.size = create_info.size;
```